lua/obsidian/config.lua

```diff
@@ -1,11 +1,14 @@
 local echo = require "obsidian.echo"
+local workspace = require "obsidian.workspace"
 
 local config = {}
 
 ---[[ Options specs ]]---
 
 ---@class obsidian.config.ClientOpts
----@field dir string
+---@field dir string|?
+---@field workspaces table
```

Comment:
I think this has the `obisidan.Workspace` type, right?
```suggestion
---@field workspaces obsidian.Workspace
```

Reply:
It's a table of `obsidian.workspace`s:

```lua
require('obsidian').setup({
    -- ...

    workspaces = {
        {
            name = 'personal',
            path = '~/Documents/notes/personal',
        },
        {
            name = 'work',
            path = '~/Documents/notes/acsd',
        },
    }
    -- ...
)}
```

Reply:
Ah, right, then we can give it a type hint like this
```suggestion
---@field workspaces obsidian.Workspace[]|?
```